lib/appstate.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:flutter/material.dart';
+
+class AppState {
+  var nickName = ValueNotifier<String>('Yanciman');
+  var userStatus = ValueNotifier<String>('Producing works of art in Kannywood');
+
+  void setNick(String nickName) {
+    this.nickName = ValueNotifier<String>(nickName);
```

Comment:
This should probably be setting `this.nickName.value` and not creating a new `ValueNotifier<String>`? Assigning to value uses the [setter](https://dart.dev/guides/language/language-tour#getters-and-setters) magic to run some code during the assignment, in this case notifying all listeners.

This also allows both members to be final.